### PR TITLE
Update `Get-AzHost` cmdlet to return Host for -ResourceId parameterset

### PR DESCRIPTION
**Pull request by pselugar**
## Description
 * Updated `Get-AzHost` cmdlet logic to return Host for `-ResouceId` ParameterSet.
 * Manually tested the changes (Debug).
 * Added recorded Test to validate changes.
 *  Incorrect path was provided for field ` String ResourceName` in the `GetResourceName()` call :  `"Microsoft.Compute/hosts"` -> `"Microsoft.Compute/hostGroups"`. Fixing this made `hostGroupName` fetch non-null value. 
 * `GetInstanceId()` is the correct function call for `hostName` instead of `GetResourceName()`  to fetch non null value for `hostName` . (Check out `DedicatedHostDeleteMethods.cs` file for reference)
* [GitHub Issue](https://github.com/Azure/azure-powershell/issues/18140)

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Compute/Compute/ChangeLog.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 ## Upcoming Release
```
Comment (Sandido) [inline src/Compute/Compute/ChangeLog.md:23]: Please link the PS design request in the PR description

File: src/Compute/Compute/ChangeLog.md [hunk 2]
```diff
@@ -22,6 +22,7 @@
...
 
 ## Version 4.32.0
+* Updated `Get-AzHost` cmdlet logic to return Host for `-ResourceId` parameterset.
```
Comment (Sandido) [inline src/Compute/Compute/ChangeLog.md:25]: Add this to the Upcoming Release section.

File: src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs [hunk 3]
```diff
@@ -48,8 +48,8 @@ public override void ExecuteCmdlet()
                 {
                     case "ResourceIdParameter":
                         resourceGroupName = GetResourceGroupName(this.ResourceId);
-                        hostGroupName = GetResourceName(this.ResourceId, "Microsoft.Compute/hosts", "hosts");
```
Comment (Sandido) [inline src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs:51]: Where is the recorded test to show this works?

Comment (Sandido) [reply src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs:51]: Explain why this change makes it work in the PR description.

File: src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs [hunk 4]
```diff
@@ -48,8 +48,8 @@ public override void ExecuteCmdlet()
...
-                        hostName = GetResourceName(this.ResourceId, "Microsoft.Compute/hosts", "hosts");
+                        hostGroupName = GetResourceName(this.ResourceId, "Microsoft.Compute/hostGroups", "hosts");
+                        hostName = GetInstanceId(this.ResourceId, "Microsoft.Compute/hostGroups", "hosts");
                         break;
```
Comment (Sandido) [inline src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs:53]: Add an example to the Get-AzHost.md file for this if it does not exist.

**Review by Sandido — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/ChangeLog.md (hunk 2 above)
Comment (pselugar) [reply src/Compute/Compute/ChangeLog.md:25]: I added there itself, but it was showing Conflict. So in order to resolve it I moved it here. Nvm will add it back to where it was.

File: src/Compute/Compute/ChangeLog.md (hunk 1 above)
Comment (pselugar) [reply src/Compute/Compute/ChangeLog.md:23]: This is a Github Issue so I was unable to find PR design.  I have linked the Github Issue though

File: src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs (hunk 3 above)
Comment (pselugar) [reply src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs:51]: I manually debug tested it, but will put up a revision with actual record testing too.

File: src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs (hunk 4 above)
Comment (pselugar) [reply src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs:53]: It does not exist.  I'll add that too

**Comment by VeryEarly** [comment]
please delete ADO workitem link in this PR

File: src/Compute/Compute/ChangeLog.md (hunk 2 above)
Comment (pselugar) [reply src/Compute/Compute/ChangeLog.md:25]: Done

File: src/Compute/Compute/ChangeLog.md (hunk 1 above)
Comment (pselugar) [reply src/Compute/Compute/ChangeLog.md:23]: Unable to find one, but already added Github Issue link

File: src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs (hunk 3 above)
Comment (pselugar) [reply src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs:51]: Added recorded tests too

Comment (pselugar) [reply src/Compute/Compute/Generated/DedicatedHost/DedicatedHostGetMethod.cs:51]: Added reasoning for working of the change.

**Comment by VeryEarly** [comment]
is this PR ready to be reviewed/merged?

**Comment by pselugar** [comment]
> is this PR ready to be reviewed/merged? Please remove the tag "Do Not Merge" if so
It's ready to be merged.
removed the "Do Not Merge" label

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**